lighthouse/classes/event_properties/interfaces.py

```diff
@@ -0,0 +1,225 @@
+from abc import ABC, abstractmethod
+from typing import Any, List, Dict
+from lighthouse.classes.messages.warehouse_messages import WarehouseMessage  # type: ignore
+from contextlib import contextmanager
+
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+class ValidationError(Exception):
+    pass
+
+
+class RetrievalError(Exception):
+    pass
+
+
+class EventPropertyInterface(ABC):
+    """
+    This class defines the public interface offered by an event property.
+    Any method we need to have present in event property should be defined
+    inside this class.
+    """
+
+    @abstractmethod
+    def validate(self) -> bool:
```

Comment:
I think this should be renamed to `is_valid` or something similar to other boolean checks.

Reply:
Yes, done